mist/master/src/main/scala/io/hydrosphere/mist/master/configs.scala

```diff
@@ -92,12 +94,74 @@ case class WorkersSettingsConfig(
   runnerInitTimeout: Duration,
   readyTimeout: FiniteDuration,
   maxArtifactSize: Long,
+  dockerConfig: DockerRunnerConfig,
+  manualConfig: ManualRunnerConfig
+)
+
+sealed trait DockerNetworkConfiguration
+case class NamedNetwork(name: String) extends DockerNetworkConfiguration
+case class AutoMasterNetwork(masterId: String) extends DockerNetworkConfiguration
+
+object DockerNetworkConfiguration {
+
+  def apply(config: Config): DockerNetworkConfiguration = {
+    config.getString("network-type") match {
+      case "auto-master" => AutoMasterNetwork(config.getConfig("auto-master-network").getString("container-id"))
+      case name => NamedNetwork(name)
+    }
+  }
+}
+
+case class DockerRunnerConfig(
   dockerHost: String,
-  dockerPort: Int,
-  cmd: String,
-  cmdStop: String
+  image: String,
+  network: DockerNetworkConfiguration,
+  mistHome: String,
+  sparkHome: String
+)
+
+object DockerRunnerConfig {
+  def apply(config: Config): DockerRunnerConfig = {
+    DockerRunnerConfig(
+      dockerHost = config.getString("host"),
+      image = config.getString("image"),
+      network = DockerNetworkConfiguration(config),
+      mistHome = config.getString("mist-home"),
+      sparkHome = config.getString("spark-home")
+    )
+  }
+}
+
+case class ManualRunnerConfig(
+  cmdStart: String,
+  cmdStop: Option[String],
+  async: Boolean
 )
 
+object ManualRunnerConfig {
+  def apply(config: Config): ManualRunnerConfig = {
+    def readOld(): ManualRunnerConfig = {
+      val stop = config.getString("cmdStop")
+      ManualRunnerConfig(
+        cmdStart = config.getString("cmd"),
+        cmdStop = if (stop.isEmpty) None else stop.some,
```

Comment:
Is this code null safe? By default, we have `cmdStop = null` and according to Config docs it will throw `ConfigException.Missing`